### PR TITLE
Fix(e2e): Fix file upload bug (bug introduced before any release happened. release note not required)

### PR DESCRIPTION
**Pull request by azabbasi**
The commit that broke the functionality:
https://github.com/Azure/azure-iot-sdk-csharp/commit/220acb35b07a1e8537f27376e1075e3504af1545

### CODE REVIEW

File: iothub/device/src/InternalClient.cs [hunk 1]
```diff
@@ -1543,20 +1543,7 @@ public Task UploadToBlobAsync(string blobName, Stream source, CancellationToken
                     throw Fx.Exception.Argument(nameof(blobName), "Path segment count cannot exceed 254");
                 }
 
-                var context = new PipelineContext();
```
Comment (azabbasi) [inline iothub/device/src/InternalClient.cs:1546]: we don't need to create a new transport handler as the internal client is initialized with one in the first place.

Comment (abhipsaMisra) [reply iothub/device/src/InternalClient.cs:1546]: was this entire block added recently?

Comment (abhipsaMisra) [reply iothub/device/src/InternalClient.cs:1546]: Could you add a bit of context on what the code was previously, and what was changed?

Comment (abhipsaMisra) [reply iothub/device/src/InternalClient.cs:1546]: Or maybe just link the commits :)

**Review by drwill-ms — APPROVED** [review]

File: iothub/device/src/InternalClient.cs [hunk 2]
```diff
@@ -1543,20 +1543,7 @@ public Task UploadToBlobAsync(string blobName, Stream source, CancellationToken
...
-                context.Set(_productInfo);
-
-                var transportSettings = new Http1TransportSettings();
```
Comment (drwill-ms) [inline iothub/device/src/InternalClient.cs:1549]: The difference between this one and the ctor one, is that the ctor could be any other transport, and this is specifically Http.

File: iothub/device/src/InternalClient.cs (hunk 1 above)
Comment (azabbasi) [reply iothub/device/src/InternalClient.cs:1546]: spoke to Abhipsa offline.

File: iothub/device/src/InternalClient.cs (hunk 2 above)
Comment (azabbasi) [reply iothub/device/src/InternalClient.cs:1549]: file upload is always over HTTP though

Comment (drwill-ms) [reply iothub/device/src/InternalClient.cs:1549]: Never mind.

Comment (azabbasi) [reply iothub/device/src/InternalClient.cs:1549]: including the one in the ctor.

**Pull request merged**